### PR TITLE
fix(github-cli-auth): surface missing GitHub auth to the agent instead of opaque failures

### PR DESCRIPTION
**Pull request by devxoul**
## Background

A typeclaw agent running inside a container on a GitHub repo could not push a branch or open a PR. Concrete symptoms:

- `gh auth status` → "You are not logged into any GitHub hosts."
- Running a push command → `fatal: could not read Username for 'https://github.com': No such device or address`

Two layered root causes:

1. **Silent pass-through on no managed auth.** The `github-cli-auth` hook (`tool.before` on `bash`) only injects a token when a `.env` PAT (`GH_TOKEN`/`GITHUB_TOKEN`) is present or a GitHub App per-repo minter is live. With neither, the hook returned early and let the command through untouched — leaving git/gh to fail with opaque, undiagnosable errors. Every *other* missing-auth case the plugin handles (sandboxed-PAT-withheld, missing-repo, multi-owner, unsafe composition, `gh api /user`) already emits block-with-guidance; the no-managed-auth case was the one remaining silent path.

2. **Agent-invisible channel misconfiguration.** When `typeclaw.json` declares a `channels.github` block but `secrets.json` carries no usable credential, the channel manager logs an operator-level error and skips the adapter entirely — the log goes to container stdout, which the LLM agent never sees. The agent only experienced the opaque git failure, with no path to learn the real cause.

## Summary

Three commits, each addressing a distinct layer:

**Commit 1 — warn, not block, for the generic no-managed-auth case.** When the hook classifies a command as repo-targeting (`inject`) but finds no managed credential, it now emits a one-shot `ctx.logger.warn` naming the situation and both remedies, then passes the command through unchanged. Why warn and not block: the agent may have configured its own auth typeclaw cannot observe (credential helper, `~/.netrc`, in-container `gh auth login`, mounted gh config, SSH remote). A hard block would break those legitimate flows. Ordering is load-bearing: the analyzer now runs *before* the no-auth early-return on the git path, so only true repo-targeting github commands warn; `gh auth status`, `git status`, and non-github remotes stay silent.

**Commit 2 (review follow-up) — fold `GITHUB_TOKEN` into the managed-auth decision.** The predicate and PAT classification keyed only off `GH_TOKEN`, so a command run with `GITHUB_TOKEN`-only falsely tripped the warning even though `gh` honors it natively. Introduced `effectiveProcessPat()` = `GH_TOKEN ?? GITHUB_TOKEN` (gh's own precedence) and keyed every process-env PAT branch off it: classification, the App-mint decision, warning suppression, the gh overlay (still injected under the higher-precedence `GH_TOKEN` name), and the git askpass token. Note the asymmetry: `gh` reads `GITHUB_TOKEN` natively but `git` does not, so folding it into the git path means the askpass helper now supplies it — making a `GITHUB_TOKEN`-only push authenticate rather than merely suppress a warning.

**Commit 3 — block the one high-confidence misconfiguration.** The only agent-visible signal from a `tool.before` hook is a block reason; a `logger.warn` is invisible to the LLM. So when `channels.github` is configured *and* secrets auth is absent *and* no env token exists *and* no live resolver is present, a credential-requiring write (push to github.com, `gh pr create`) is now blocked with the exact root cause and remedy instead of failing opaquely. This is the only case blocked: here — unlike the generic no-managed-auth case — typeclaw *knows* auth is both intended and missing.

## What this does NOT do

- Does not let an agent push without credentials — it converts an undiagnosable failure into an actionable, agent-visible one.
- Does not change any path where a PAT or GitHub App auth is configured — all existing inject/mint/block paths are unchanged.
- Does not block reads or ambiguous cases: only pushes to github.com and `gh pr create` block, and only in the configured-without-credentials state. `clone`/`fetch`/`pull`/`gh pr view`/`gh auth status`/non-github remotes/`--help` invocations are never blocked.
- Does not false-block on schema-invalid or unreadable `secrets.json` (treated as indeterminate) or when a live App resolver exists (multi-owner/no-repos App auth bypasses the path entirely).
- Does not touch the config/secrets schema, Dockerfile, or token-bridge wiring. The new detector reads `typeclaw.json`/`secrets.json` for presence/absence only — never token values.

## Review notes

- **Load-bearing changes:** the two no-managed-auth sites in `src/bundled-plugins/github-cli-auth/index.ts` (`handleGhCommand`/`handleGitCommand`) now route through a shared `resolveNoManagedAuth(isCredentialRequiringWrite: boolean)`; new `channel-misconfig.ts` is a memoized, per-agentDir state detector (presence-only); new predicates `isGitPushCommand` / `isGhCredentialRequiringWrite` both exclude `--help`/`-h`; `effectiveProcessPat` in `token-class.ts`.
- **Invariant to verify:** the misconfig block can only fire for `inject`-classified commands after the block/pass-through/PAT-inject branches have returned; a live App resolver bypasses it; only `configured-without-credentials` blocks — `indeterminate`/`with-credentials`/`not-configured` never block.
- **Tests:** full `github-cli-auth` suite (369 cases) green — adds `GITHUB_TOKEN`-only coverage for gh and git, precedence over `GH_TOKEN`, sandboxed `GITHUB_TOKEN`-only paths, the configured-without-credentials block and all the not-blocked guards (reads, non-github, `--help`, valid creds, no `channels.github`, malformed `secrets.json`), plus unit tests for the detector and predicates.

### CODE REVIEW

File: src/bundled-plugins/github-cli-auth/index.ts [hunk 1]
```diff
@@ -160,7 +183,13 @@ export default definePlugin({
       // No App auth (no App-class GH_TOKEN and no live minter): leave whatever
       // is seeded so `gh` fails honestly rather than us guessing a token. The
       // sandboxed-PAT mint path bypasses this PAT-class re-check via the flag.
-      if (!mintForSandboxedPat && !shouldMintAppToken(process.env.GH_TOKEN, hasAppTokenResolver())) return
+      // `decision` is necessarily `inject` here (block/pass-through returned
+      // above), i.e. a repo-targeting gh with no managed credential — warn so the
+      // pass-through is diagnosable, then let it run unchanged.
+      if (!mintForSandboxedPat && !shouldMintAppToken(process.env.GH_TOKEN, hasAppTokenResolver())) {
```
[the remark this answers is not in the document]
Comment (devxoul) [reply src/bundled-plugins/github-cli-auth/index.ts]: Fixed in 481e7146. Root of the false positive was that the no-managed-auth predicate (and the PAT classification feeding it) only read `process.env.GH_TOKEN`.

Took the "fold it in" route: added `effectiveProcessPat()` = `GH_TOKEN ?? GITHUB_TOKEN` (gh's own precedence) and keyed **every** process-env PAT branch off that one named value — classification, the App-mint decision, warning suppression, the unsandboxed `gh` overlay (still injected under the higher-precedence `GH_TOKEN` name), and the unsandboxed `git` askpass token.

Note on the gh-vs-git asymmetry you implicitly surfaced: `gh` honors `GITHUB_TOKEN` natively but `git` does not, so for the git path folding it in means the askpass helper now actually *supplies* it — i.e. `GITHUB_TOKEN`-only `git push` authenticates instead of just having its warning suppressed. The single named local also closes the `gitToken = process.env.GH_TOKEN` → `undefined` trap. Sandbox behavior is unchanged (a `GITHUB_TOKEN` PAT is equally cleared by `--clearenv`, so it follows the same withheld → App-mint-preferred → block-with-guidance path).

Regression coverage added: `GITHUB_TOKEN`-only repo-targeting gh + git (no false warning, correct injection), `GH_TOKEN`-over-`GITHUB_TOKEN` precedence on both paths, sandboxed `GITHUB_TOKEN`-only (App mint / withheld block), and unit tests for `effectiveProcessPat`.

**Comment by devxoul** [comment]
## Follow-up: addressing the actual core issue (30d6ffba)

The earlier commits made the no-managed-auth failure *legible* (a warning) but did not address what the reviewer/user correctly called out as the **core issue**: the agent still couldn't push, and — more importantly — the diagnostic was emitted to the **wrong audience**.

### Root cause (re-stated precisely)
When `typeclaw.json` has a `channels.github` block but `secrets.json#channels.github` has no usable auth, the channel manager logs an operator-level error (`[channels] adapter "github" missing credentials; skipping`) and never starts the adapter — so no token resolver is live. **That log goes to container stdout; the agent doing the work never sees it.** The agent only experiences GitHub's opaque `could not read Username`, with no path to learn the real cause.

The credential genuinely doesn't exist, so the fix can't be "mint a token." The fix is to make the diagnosis **reach the actor**.

### What changed
The only agent-visible channel from a `tool.before` hook is a **block reason** (a `logger.warn` is invisible to the LLM session). So for the ONE high-confidence misconfiguration — `channels.github` configured **+** secrets auth absent **+** no env token **+** no live resolver — a **credential-requiring write** (`git push`, `gh pr create`) is now blocked with the exact root cause and fix, instead of failing opaquely.

### Scope guards (to avoid false-blocking ambient auth)
- Only `git push` over a github.com remote and `gh pr create` block. `clone`/`fetch`/`pull`/`gh pr view`/`gh auth status`/non-github remotes keep the generic warn-and-pass-through.
- `--help`/`-h` excluded (no credential needed).
- Only the `configured-without-credentials` state blocks. Schema-invalid/unreadable secrets → `indeterminate` → never false-blocks. Valid auth, or no github channel, never hit this path.
- A live App resolver (multi-owner / no-repos App auth) bypasses the no-auth branch entirely → unaffected.

### Implementation
- New `channel-misconfig.ts`: detects state by **presence/absence only** (never reads token values), memoized per agentDir (config/secrets are restart-required).
- New write-op predicates `isGitPushCommand` / `isGhCredentialRequiringWrite`.
- Integration + detector-unit + predicate-unit tests (369 in the plugin suite, all green). Design + a self-review pass done with the architecture-review agent; the one edge case it flagged (`--help` false-block) is fixed with regression coverage.

Honest note: this does not let an agent push *without* credentials — nothing can. It converts an undiagnosable failure into an actionable, agent-visible one that names the exact missing config + remedy.

**Comment by devxoul** [comment]
Closing per maintainer decision. This PR only *surfaced* the missing-auth condition (warn / misconfig-block) — it was not the core fix. The actual root cause is that built-in subagents never ran the github-cli-auth tool.before hook, so token minting never reached their bash; that is fixed in #876. Closing this to avoid a redundant/competing diagnostic layer.
